### PR TITLE
Add pressure transducer documentation

### LINKED ISSUE #627: Document Pressure Transducers
We should add some documentation about pressure transducers, what they do, and how to write code for them (AnalogInput & scale).

### PR DESCRIPTION
**Pull request by Starlight220**
Resolves #627 .
Probably needs some fixing with the API etc, reviews will help.

### CODE REVIEW

File: source/docs/software/actuators/pneumatics.rst [hunk 1]
```diff
@@ -112,3 +112,24 @@ Double solenoids are controlled by the DoubleSolenoid class in WPILib. These are
         exampleDouble.Set(frc::DoubleSolenoid::Value::kOff);
         exampleDouble.Set(frc::DoubleSolenoid::Value::kForward);
         exampleDouble.Set(frc::DoubleSolenoid::Value::kReverse);
+
+
+Pressure Sensors
+----------------
+
+To read the pressure stored in the pneumatic system, connecting an analog pressure sensor is possible. These sensors connect to the Analog ports on the roboRIO, and can be read by the AnalogInput class in WPILib.
```
Comment (prateekma) [inline source/docs/software/actuators/pneumatics.rst:120]: Consider rewording the first sentence to "One can connect a pressure sensor to measure the pressure stored in a pneumatic system."

File: source/docs/software/actuators/pneumatics.rst [hunk 2]
```diff
@@ -112,3 +112,24 @@ Double solenoids are controlled by the DoubleSolenoid class in WPILib. These are
...
+.. tabs:
+
+    .. code-tab:: java
+        import edu.wpi.first.wpilibj.AnalogInput;
+
+        AnalogInput pressureSensor = new AnalogInput(/* the AnalogIn port*/ 2);
+
+        //scaled units in the 0V-5V range
```
Comment (prateekma) [inline source/docs/software/actuators/pneumatics.rst:128]: Add space between `//` and start of comment

File: source/docs/software/actuators/pneumatics.rst [hunk 3]
```diff
@@ -112,3 +112,24 @@ Double solenoids are controlled by the DoubleSolenoid class in WPILib. These are
...
+        double volts = pressureSensor.getVoltage();
+
+    .. code-tab:: c++
+        frc::AnalogInput pressureSensor {2};
+
+        //scaled units in the 0V-5V range
```
Comment (prateekma) [inline source/docs/software/actuators/pneumatics.rst:134]: Add space between `//` and start of comment

File: source/docs/software/actuators/pneumatics.rst [hunk 4]
```diff
@@ -112,3 +112,24 @@ Double solenoids are controlled by the DoubleSolenoid class in WPILib. These are
...
+    .. code-tab:: java
+        import edu.wpi.first.wpilibj.AnalogInput;
+
+        AnalogInput pressureSensor = new AnalogInput(/* the AnalogIn port*/ 2);
+
+        //scaled units in the 0V-5V range
+        double volts = pressureSensor.getVoltage();
+
+    .. code-tab:: c++
+        frc::AnalogInput pressureSensor {2};
```
Comment (prateekma) [inline source/docs/software/actuators/pneumatics.rst:132]: Remove space between `pressureSensor` and `{2}`

**Comment by sciencewhiz** [comment]
While pressure sensor is technically correct, I think it may be confusing with pressure switch. For that reason, I prefer the term pressure transducer. 

Add a reference to look at the sensor data sheet for how to turn voltage into pressure. While some are linear, I've used ones that aren't.

File: source/docs/software/actuators/pneumatics.rst (hunk 1 above)
Comment (sciencewhiz) [reply source/docs/software/actuators/pneumatics.rst:120]: Specify Analog Input port On RoboRIO, since there are also Analog Output ports

Comment (Starlight220) [reply source/docs/software/actuators/pneumatics.rst]: Fixed both.

File: source/docs/software/actuators/pneumatics.rst (hunk 4 above)
Comment (Starlight220) [reply source/docs/software/actuators/pneumatics.rst]: It was consistent with the other code samples. Should they also be changed?

Comment (Daltz333) [reply source/docs/software/actuators/pneumatics.rst]: What other code examples?

**Comment by Daltz333** [comment]
The error is currently due to invalid code block syntax

**Comment by Starlight220** [comment]
> Add a reference to look at the sensor data sheet for how to turn voltage into pressure. While some are linear, I've used ones that aren't.

Explain? I understood that it should be documented that teams should implement a function for `Volts -> PSI`, but something more concrete would be helpful.

File: source/docs/software/actuators/pneumatics.rst (hunk 4 above)
Comment (Starlight220) [reply source/docs/software/actuators/pneumatics.rst]: The DoubleSolenoid, Solenoid, and Comperssor code examples. Literally every other code sample on the page.

**Comment by sciencewhiz** [comment]
For example, this sensor provides the formula in the datasheet https://www.revrobotics.com/content/docs/REV-11-1107-DS.pdf
p = 250 * (V / 5.0) - 25

But this one does not provide a formula, but does provide a graph https://www.ebay.com/itm/Pressure-transducer-sender-150-psi-5V-stainless-steel-for-oil-fuel-air-water-/251362960044 from which you can derive a formula
p = 37.5 * (V - 0.5)

Fundamentally those are the same equation, just scaled for the maximum pressure of 200 PSI for the former and 150 PSI for the latter. However, other transducers may use other sensors and not have the same scaling or equation.

I was wrong when I said we used one that wasn't linear.

**Comment by Starlight220** [comment]
So to add a concrete example featuring one of the sensors, or a general saying of "there's a formula, use it."
Anyway, what's the API for scaling/shifting AnalogInputs?

**Comment by Starlight220** [comment]
thanks Austin for fixing my bad merge

File: source/docs/software/actuators/pneumatics.rst [hunk 5]
```diff
@@ -112,3 +112,38 @@ Double solenoids are controlled by the DoubleSolenoid class in WPILib. These are
...
+        double psi = pressureTransducer.get();
+
+    .. code-tab:: c++
+
+        // product-specific voltage->pressure conversion, see product manual
+        // in this case, 250(V/5)-25
+        // the scale parameter in the AnalogPotentiometer constructor is scaled from 1 instead of 5,
+        // so if r is the raw AnalogPotentiometer output, the pressure is 250r-25
+        double scale = 250, offset = -25;
+        frc::AnalogPotentiometer pressureTransducer {/* the AnalogIn port*/ 2, scale, offset};
```
Comment (prateekma) [inline source/docs/software/actuators/pneumatics.rst:146]:
```suggestion
        frc::AnalogPotentiometer pressureTransducer{/* the AnalogIn port*/ 2, scale, offset};
```

Comment (Starlight220) [reply source/docs/software/actuators/pneumatics.rst:146]: @Daltz333 - what do you say about this change? All other cpp code is with 
```cpp
namespace::Type variableName {args}
```
format. 
If we want to remove the space before the arg braces, it should be done for all code on the page or even all code in frc-docs.

Comment (Daltz333) [reply source/docs/software/actuators/pneumatics.rst:146]: That is correct. Code should follow the WPILib style guide.

Comment (Starlight220) [reply source/docs/software/actuators/pneumatics.rst:146]: And wpiformat has no space before args?

**Pull request merged**